pkg/controllers/jobset_controller.go

```diff
@@ -916,3 +914,42 @@ func findReplicatedStatus(replicatedJobStatus []jobset.ReplicatedJobStatus, repl
 	}
 	return jobset.ReplicatedJobStatus{}
 }
+
+// messageWithFirstFailedJob appends the first failed job to the original event message in human readable way.
```

Comment:
Could we move these out of this file?

They don't require the controller so it may be useful to move them to a separate file.

Reply:
Yeah jobset_controller.go is overdue for refactoring. To start I think we can refactor some of the many helper functions into separate files based on the feature (similar to what you did with startup policy). 

I did some refactoring in this PR (e.g. moving some functions into success_policy.go, adding a constants pkg, etc.)

However, for these particular functions, I'm not sure of the best place to put them yet. They are mostly about finding the first failed job for a Jobset, which doesn't lend itself to a cleanly separated feature file.